tests/defences/trainer/test_adversarial_trainer_trades_pytorch.py

```diff
@@ -47,6 +51,46 @@ def _get_adv_trainer():
                 verbose=False,
             )
             trainer = AdversarialTrainerTRADESPyTorch(classifier, attack, beta=6.0)
+
+        if framework == "huggingface":
+
+            if HF_MODEL_SIZE == "LARGE":
+                import transformers
+                import torch
+                from art.estimators.hugging_face import HuggingFaceClassifier
+
+                model = transformers.AutoModelForImageClassification.from_pretrained(
+                    "facebook/deit-tiny-patch16-224", ignore_mismatched_sizes=True, num_labels=10  # takes 3 min
+                )
+
+                print("num of parameters is ", sum(p.numel() for p in model.parameters() if p.requires_grad))
+                optimizer = torch.optim.Adam(model.parameters(), lr=1e-4)
+
+                classifier = HuggingFaceClassifier(
+                    model,
+                    loss=torch.nn.CrossEntropyLoss(),
+                    input_shape=(3, 224, 224),
+                    nb_classes=10,
+                    optimizer=optimizer,
+                    processor=None,
+                )
+            elif HF_MODEL_SIZE == "SMALL":
+                classifier = get_image_classifier_hf()
+
+            attack = ProjectedGradientDescent(
+                classifier,
+                norm=np.inf,
+                eps=0.3,
+                eps_step=0.03,
+                max_iter=5,
+                targeted=False,
+                num_random_init=1,
+                batch_size=128,
+                verbose=False,
+            )
+
+            trainer = AdversarialTrainerTRADESPyTorch(classifier, attack, beta=6.0)
+
```

Comment:
```suggestion
        if framework == "huggingface":
            trainer = None
```

Reply:
Could we call `image_dl_estimator` instead?